packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -0,0 +1,368 @@
+
+    this.handleDaylightSavingButtonClick = this.handleDaylightSavingButtonClick.bind(this);
+    this.handleStandardTimeButtonClick = this.handleStandardTimeButtonClick.bind(this);
+  }
+
+  componentWillMount() {
+    const dateFormat = DateUtil.getFormatByLocale(this.context.intl.locale);
+
+    this.dateValue = DateTimeUtils.formatMomentDateTime(this.state.dateTime, dateFormat);
+    this.timeValue = DateTimeUtils.hasTime(this.props.value) ? DateTimeUtils.formatISODateTime(this.props.value, 'HH:mm') : '';
+  }
+
+  handleOnSelect(event, selectedDate) {
+    const dateFormat = DateUtil.getFormatByLocale(this.context.intl.locale);
+    this.dateValue = DateTimeUtils.formatISODateTime(selectedDate, dateFormat);
+    const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+    const updatedDateTime = DateTimeUtils.syncDateTime(previousDateTime, selectedDate, this.timeValue);
+
+    if (!previousDateTime || previousDateTime.format() !== updatedDateTime.format()) {
+      this.checkAmbiguousTime(updatedDateTime);
+    }
+  }
+
+  handleOnDateBlur(event) {
+    const dateFormat = DateUtil.getFormatByLocale(this.context.intl.locale);
+    const isDateTimeValid = DateTimeUtils.isValidDateTime(event.target.value, this.timeValue, dateFormat);
+    const enteredDateTime = isDateTimeValid ? this.state.dateTime : null;
+
+    this.checkAmbiguousTime(enteredDateTime);
+  }
+
+  handleOnTimeBlur() {
+    const dateFormat = DateUtil.getFormatByLocale(this.context.intl.locale);
+    const isDateTimeValid = DateTimeUtils.isValidDateTime(this.dateValue, this.timeValue, dateFormat);
+    let updatedDateTime;
+
+    if (isDateTimeValid) {
+      updatedDateTime = DateTimeUtils.updateTime(this.state.dateTime, this.timeValue);
+    }
+
+    this.checkAmbiguousTime(updatedDateTime);
+  }
+
+  checkAmbiguousTime(dateTime) {
+    // To prevent multiple time clarification dialogs from rendering, ensure that it is not open before checking for the ambiguous hour.
+    // One situation is when using the right arrow key to move focus from the hour input to the minute input, it will invoke onBlur and check for ambiguous hour.
+    // If the hour is ambiguous, the dialog would display and steal focus from the minute input, which again will invoke onBlur and check for ambiguous hour.
+    if (this.state.isTimeClarificationOpen) {
+      return;
+    }
+
+    let isDateTimeAmbiguous = false;
+    const isOldTimeAmbiguous = this.state.isAmbiguousTime;
+    if (dateTime && dateTime.isValid()) {
+      const tempDateTime = dateTime.clone();
+      isDateTimeAmbiguous = DateTimeUtils.checkAmbiguousTime(tempDateTime);
+    }
+
+    this.setState({
+      isAmbiguousTime: isDateTimeAmbiguous,
+      isTimeClarificationOpen: isDateTimeAmbiguous && !isOldTimeAmbiguous,
+    });
+  }
+
+  handleDateChange(event, date) {
+    if (event.type === 'change') {
+      this.dateValue = event.target.value;
```

Comment:
should we continue on if the event is not 'change'?  why not just return right away?

Reply:
When a date is selected from the picker, as opposed to manually entering a date, the event.type is 'click' and this function would still need to get executed to handle the date change. However, on selecting a date from the picker, event.target.value is null so it should 
 not overwrite this.dateValue that is already set in handleOnSelect.